### ISSUE TITLE
"dict attributes" honour neither `literal_false` nor `default`

### ISSUE THREAD

**Issue by d-maurer** [malthe/chameleon#316]
In the python output below, both templates should produce equivalent output - but they don't.
```
>>> from chameleon.zpt.template import PageTemplate
>>> from sys import intern
>>> default = intern('__default__')
>>> PageTemplate('<div d="orig" tal:define="d dict(x=1, f=False, n=None, d=default)" tal:attributes="d"/>')(default=default)
'<div x="1" f="False" d="__default__"/>'
>>> PageTemplate('<div d="orig" tal:attributes="x 1; f False; n None; d default"/>')(default=default)
'<div d="orig" x="1"/>'
```

Python 3; chameleon==3.7.4

**Comment by malthe** [comment]
The internal default marker is not exposed or usable from the outside. It's used as a non-value. But it makes me wonder why `default` is not just set to the actual default value.

**Comment by d-maurer** [comment]
Malthe Borch wrote at 2020-6-21 00:47 -0700:
>The internal default marker is not exposed or usable from the outside.

The original TAL/TALES specification had defined the TALES "default" value
and its meaning: keep the value from the template and do not override it
by the dynamically computed value.

Thus, `chameleon` should clearly provide such a value to be used for this
purpose. Ideally, `chameloon` should put this value automatically
into the TALES context -- but it does not do this. That's why
the example code uses internal implementation details to provide
`default`.

At this place, it is only relevant that it works for
"simple" attributes but not for "dict attribute"s.


I need a TALES `default` for the new integration of `chameleon` with Zope
(using `zope.tales` in place of `chameleon.tales` as TALES implementation.
We have discussed this elsewhere.

**Comment by d-maurer** [comment]
Malthe Borch wrote at 2020-6-21 00:47 -0700:
>The internal default marker is not exposed or usable from the outside. It's used as a non-value. But it makes me wonder why `default` is not just set to the actual default value.

I suspect strongly, that "boolean attrutes", too,
are not handled correctly if defined via a "dict attributes".

**Comment by malthe** [comment]
I can confirm this. Dictionary attributes ignore the `literal_false` setting.

**Comment by d-maurer** [comment]
Malthe Borch wrote at 2020-6-22 08:54 -0700:
>I can confirm this. Dictionary attributes ignore the `literal_false` setting.

and almost surely "default" and boolean attributes, too.

**Comment by malthe** [comment]
I believe this has been fixed with #382.
